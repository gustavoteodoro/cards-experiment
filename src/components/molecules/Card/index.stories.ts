import Card from './index.vue';

export default { title: 'Molecules/Card' };

export const Default = () => ({
  components: { Card },
  template: '<Card value="2" suit="SPADES" />'
});