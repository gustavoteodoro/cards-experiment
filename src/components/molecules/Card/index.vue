<template>
  <div class="container">
    <CardCode :value="value" :suit="suit" />
    <CardCode :value="value" :suit="suit" inverted />
  </div>
</template>

<script>
import CardCode from '../../atoms/CardCode/index.vue';

export default {
  name: 'Card',
  props: {
    value: String,
    suit: String,
  },
  components: { CardCode },
};
</script>

<style scoped>
.container{
  background: white;
  width: 100%;
  display: flex;
  flex-direction: column;
  max-width: 136px;
  min-height: 240px;
  justify-content: space-between;
  box-shadow: 1px 1px 6px 1px #9686a3;
  border-radius: 15px;
  box-sizing: border-box;
  padding: 7px 18px;
}
</style>
