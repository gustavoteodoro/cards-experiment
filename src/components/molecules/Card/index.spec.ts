import { mount } from "@vue/test-utils";
import Card from "./index.vue"

describe("Card", () => {
  it('renders correctly King', () => {
    const wrapper = mount(Card , {
      propsData: {
        value: "KING",
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
