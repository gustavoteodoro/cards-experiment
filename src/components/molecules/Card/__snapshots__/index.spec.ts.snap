// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly King 1`] = `
<div
  class="container"
>
  <div
    style="transform: rotate(0deg);"
  >
    <span>
      K
    </span>
     
    <img />
  </div>
   
  <div
    style="transform: rotate(180deg);"
  >
    <span>
      K
    </span>
     
    <img />
  </div>
</div>
`;
